### PR TITLE
api

### PR DESCRIPTION
**Pull request by egdegd**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/linkCut", api.PostLinkCut).Methods(http.MethodPost)
+	router.HandleFunc("/customLink", api.PostCustomLink).Methods(http.MethodPost)
+	router.HandleFunc("/login", api.PostLogin).Methods(http.MethodPost)
+	router.HandleFunc("/register", api.PostRegister).Methods(http.MethodPost)
+	router.HandleFunc("/links", api.GetLinks).Methods(http.MethodGet)
+	router.HandleFunc("/deleteLink", api.PostDeleteLink).Methods(http.MethodPost)
```
Comment (M0nsteR) [inline main.go]: Please use another HTTP verb here (DELETE) to follow the principles of REST. Reuse the route "/links" for that functionality.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	router.HandleFunc("/changeAddress", api.PostChangeAddress).Methods(http.MethodPost)
```
Comment (M0nsteR) [inline main.go]: Please consider using either PUT or PATCH verb here. Again, this is a modifying behaviour for "/links".

File: main.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"URezL/api"
+)
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/linkCut", api.PostLinkCut).Methods(http.MethodPost)
```
Comment (M0nsteR) [inline main.go]: REST APIs usually make use of a single entity route like "/links" to implement all CRUD operations with it. Please consider simplifying your set of routes.

**Review by M0nsteR — CHANGES_REQUESTED** [review]

File: main.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/links/cut", api.PostLinkCut).Methods(http.MethodPost)
+	router.HandleFunc("/links/custom", api.PostCustomLink).Methods(http.MethodPost)
+	router.HandleFunc("/login", api.PostLogin).Methods(http.MethodPost)
+	router.HandleFunc("/register", api.PostRegister).Methods(http.MethodPost)
+	router.HandleFunc("/links", api.GetLinks).Methods(http.MethodGet)
+	router.HandleFunc("/links", api.PostDeleteLink).Methods(http.MethodDelete)
```
Comment (M0nsteR) [inline main.go]: POST or DELETE? these are separate HTTP verbs.

File: main.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	err := server.ListenAndServe()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getTest(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	o := struct {
+		Hello string `json:"hello"`
```
Comment (M0nsteR) [inline main.go]: Come on, we're all adults here)) Can you think of a better test case?

File: api/handlerFuncs.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type Link struct {
+	Link string `json:"link"`
```
Comment (M0nsteR) [inline api/handlerFuncs.go:24]: Is this enough? Just a link. No full or short form, no reference to its owner/account, no nothing?

**Review by M0nsteR — CHANGES_REQUESTED** [review]

File: api/handlerFuncs.go (hunk 6 above)
Comment (evgenyim) [reply api/handlerFuncs.go:24]: This method would shorten link in some way, which user don't know. For customizing links we have special method

Comment (M0nsteR) [reply api/handlerFuncs.go:24]: I mean, are you going to save the resulting short link somewhere, in some kind of storage? For that you need some struct field, I guess.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Some issues are still not resolved..

File: main.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/links/cut", api.PostLinkCut).Methods(http.MethodPost)
+	router.HandleFunc("/links/custom", api.PostCustomLink).Methods(http.MethodPost)
+	router.HandleFunc("/login", api.PostLogin).Methods(http.MethodPost)
+	router.HandleFunc("/register", api.PostRegister).Methods(http.MethodPost)
+	router.HandleFunc("/links", api.GetLinks).Methods(http.MethodGet)
+	router.HandleFunc("/links", api.DeleteLink).Methods(http.MethodDelete)
```
Comment (NekoNe) [inline main.go]: Sure, it is a legal way to use a delete method. But it doesn't fit in a Rest style. We are deleting resource. So, we must provide resource via URL. So, what about links identification?

**Review by M0nsteR — APPROVED** [review]
Let's merge this, but you still need to refactor your code. Please follow the file layout of our example chat project.

**Pull request merged**